.github/workflows/ci_linux_arm_mu4.yml

```diff
@@ -0,0 +1,248 @@
+        SENTRY_PROJECT=${{ github.event.inputs.sentry_project }}
+        SENTRY_URL=""
+        if [ "$SENTRY_PROJECT" == "editor" ] && [ ${{ secrets.SENTRY_SERVER_MU3_KEY }} != 0 ]; then 
+          DO_UPLOAD_SYMBOLS='true'
+          SENTRY_URL=https://sentry.musescore.org/api/2/minidump/?sentry_key=${{ secrets.SENTRY_SERVER_MU3_KEY }}
+        fi
+
+        if [ "$SENTRY_PROJECT" == "sandbox" ] && [ ${{ secrets.SENTRY_SERVER_SANDBOX_KEY }} != 0 ]; then 
+          DO_UPLOAD_SYMBOLS='true'
+          SENTRY_URL=https://sentry.musescore.org/api/3/minidump/?sentry_key=${{ secrets.SENTRY_SERVER_SANDBOX_KEY }}
+        fi
+
+        if [ $DO_BUILD == 'false' ]; then
+          DO_UPLOAD_SYMBOLS='false'
+          DO_PUBLISH='false'
+        fi
+
+        ADD_INFO="_${GITHUB_REF#refs/heads/}"
+        if [ "${{ github.event_name }}" == "schedule" ] && [ "${{ github.event.schedule }}" == "0 5 */1 */1 *" ]; then ADD_INFO="_4.0.1"; fi
+        if [ "${{ github.event_name }}" == "pull_request" ]; then ADD_INFO="_${{ github.event.pull_request.number }}_${pull_request_title}"; fi
+        UPLOAD_ARTIFACT_NAME="$(tr '":<>|*?/\\' '_' <<<"MU4_${BUILD_NUMBER}_Lin_${ADD_INFO}")"
+
+        echo "github.repository: ${{ github.repository }}"
+        echo "BUILD_MODE=$BUILD_MODE" >> $GITHUB_ENV
+        echo "BUILD_MODE: $BUILD_MODE"
+        echo "BUILD_NUMBER=$BUILD_NUMBER" >> $GITHUB_ENV
+        echo "BUILD_NUMBER: $BUILD_NUMBER"
+        echo "DO_BUILD=$DO_BUILD" >> $GITHUB_ENV
+        echo "DO_BUILD: $DO_BUILD"
+        echo "DO_PLACEHOLDER_TRANSLATIONS=$DO_PLACEHOLDER_TRANSLATIONS" >> $GITHUB_ENV
+        echo "DO_PLACEHOLDER_TRANSLATIONS: $DO_PLACEHOLDER_TRANSLATIONS"
+        echo "DO_PUBLISH=$DO_PUBLISH" >> $GITHUB_ENV
+        echo "DO_PUBLISH: $DO_PUBLISH"
+        echo "DO_UPLOAD_SYMBOLS=$DO_UPLOAD_SYMBOLS" >> $GITHUB_ENV
+        echo "DO_UPLOAD_SYMBOLS: $DO_UPLOAD_SYMBOLS"
+        echo "SENTRY_PROJECT=$SENTRY_PROJECT" >> $GITHUB_ENV
+        echo "SENTRY_PROJECT: $SENTRY_PROJECT"
+        echo "SENTRY_URL=$SENTRY_URL" >> $GITHUB_ENV
+        echo "SENTRY_URL: $SENTRY_URL"
+        echo "UPLOAD_ARTIFACT_NAME=$UPLOAD_ARTIFACT_NAME" >> $GITHUB_ENV
+        echo "UPLOAD_ARTIFACT_NAME: $UPLOAD_ARTIFACT_NAME"
+
+    - name: Setup, Generate, Update, Build the AppImage
+      if: env.DO_BUILD == 'true'
+      run: |
+        YT_API_KEY=${{ secrets.YOUTUBE_API_KEY }}; if [ -z "$YT_API_KEY" ]; then YT_API_KEY="''"; fi
+        C_URL=${SENTRY_URL}; if [ -z "$C_URL" ]; then C_URL="''"; fi
+        sudo docker run -i -v "${PWD}:/MuseScore" "arm64v8/ubuntu:bionic" /bin/bash -c "cd /MuseScore && \
+        /MuseScore/build/ci/linux_ARM/setup.sh --arch aarch64 && \
+        /MuseScore/build/ci/translation/run_lrelease.sh && \
+        /MuseScore/build/ci/linux/build.sh -n ${{ env.BUILD_NUMBER }} --youtube_api_key $YT_API_KEY --crash_log_url $C_URL && \
+        /MuseScore/build/ci/linux/package.sh && \
+        /MuseScore/build/ci/tools/checksum.sh"
```

Comment:
It's better to take separate steps, so it's easier to figure out when something breaks

Reply:
The problem is that all the steps are run inside a Docker container for emulating an ARM machine. The steps are separated, but we need to launch them sequentially in the same container in order to avoid the need to reinstall all the dependencies (it takes some time).

Maybe we could create a Docker image with all the needed dependencies and compiled binary tools (linuxdeploy, appimagetool, et cetera), but in this script we almost completetly using "official" sources (only the Qt ppa could be an exception); with a custom Docker image we would hard-code an address to a personal Docker account.

Reply:
Oh, I get it